### PR TITLE
Mid 129 donation details admin panel

### PR DESCRIPTION
**Pull request by BigTako**
Ticket: [this](https://trello.com/c/7zazoLdK/129-%D0%B0%D0%BF-%D0%B4%D0%BE%D0%BD%D0%B0%D1%82%D0%B8)
Deploy: https://mayak-rouge.vercel.app (login: admin, password: admin)

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -314,3 +314,24 @@ model WorkTime {
   @@unique([weekDay, time, isDayOff])
   @@map("work_time")
 }
+
+model DonationDetails {
+  id                   String  @id @default(uuid()) @db.Uuid
+  donationEnabled      Boolean @default(true)
+  title                String  @unique @db.VarChar(50)
+  subtitle             String  @db.VarChar(50)
+  subtitleEnabled      Boolean @default(true)
```
Comment (alex-popov-tech) [inline prisma/schema.prisma]: 🟠 for all these `${fieldname}Enabled` - wouldn't be safe to assume if there is null | '' - it is disabled?

File: src/app/(admin)/admin/_components/DonationDetails/Show.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { BooleanField, FunctionField, Show, SimpleShowLayout, TextField } from 'react-admin';
+import { QRCodeSVG } from 'qrcode.react';
+
+// model DonationDetails {
```
Comment (alex-popov-tech) [inline src/app/(admin)/admin/_components/DonationDetails/Show.js]: 🟢 forgot to remove

**Review by alex-popov-tech — APPROVED** [review]

File: prisma/data.mjs [hunk 3]
```diff
@@ -569,3 +569,21 @@ export const psychologyMethods = [
     title: 'Інше',
   },
 ];
+
+export const donationDetails = {
+  donationEnabled: true,
```
Comment (cherkasovnew) [inline prisma/data.mjs]: This is not a strict rule, but the naming of Boolean variables should start with `is...`, `isDonationEnabled `

File: prisma/schema.prisma [hunk 4]
```diff
@@ -314,3 +314,24 @@ model WorkTime {
...
+  paypalLink           String  @db.Text
+  paypalLinkEnabled    Boolean @default(true)
+  privatLink           String  @db.Text
+  privatLinkEnabled    Boolean @default(true)
+  bankDetailsEnabled   Boolean @default(true)
+  enterpriceName       String  @db.VarChar(128) // Назва підприємства
+  enterpriseRegisterId String  @db.VarChar(20) // ЄДРПОУ
```
Comment (cherkasovnew) [inline prisma/schema.prisma]: if ЄДРПОУ can only be as a number, perhaps it’s worth indicating the `Number`.

File: src/app/(admin)/admin/_lib/validationSchemas/donationDetailsSchema.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  paypalLink: zUrl,
+  paypalLinkEnabled: zBoolean,
+  privatLink: zUrl,
+  privatLinkEnabled: zBoolean,
+  bankDetailsEnabled: zBoolean,
+  enterpriceName: zString,
+  iban: zString.refine(value => iban.isValid(value), {
+    message: 'Введіть коректний IBAN',
+  }),
+  enterpriseRegisterId: z.string().refine(value => value.length === 8 && !Number.isNaN(Number(value)), {
```
Comment (cherkasovnew) [inline src/app/(admin)/admin/_lib/validationSchemas/donationDetailsSchema.js]: Could you move this value to a separate const `REGISTER_ID_LENGTH = 8`?

`...refine(value => value.length === REGISTER_ID_LENGTH ...`

` message: 'Код ЄДРПОУ має містити тільки цифри та мати довжину ${REGISTER_ID_LENGTH} символів',`

File: prisma/schema.prisma (hunk 1 above)
Comment (BigTako) [reply prisma/schema.prisma]: Thanks for comment! Yep, those fields unavailable to be null. Allowing them to be null causes extra conditions on front-end(`${fieldname}Enabled &&...` ), which provides the same result with more code, so i found first variant to be better. Moreover on the admin panel, those fields are Boolean select , so they are always true or false.

File: src/app/(admin)/admin/_components/DonationDetails/Show.js (hunk 2 above)
Comment (BigTako) [reply src/app/(admin)/admin/_components/DonationDetails/Show.js]: Thanks for comment! Removed.

File: src/app/(admin)/admin/_lib/validationSchemas/donationDetailsSchema.js (hunk 5 above)
Comment (BigTako) [reply src/app/(admin)/admin/_lib/validationSchemas/donationDetailsSchema.js]: Thanks for comment! Yep, seems better. Moved  `REGISTER_ID_LENGTH` to separate const variable.

File: prisma/data.mjs (hunk 3 above)
Comment (BigTako) [reply prisma/data.mjs]: Thanks for comment! Renamed.

File: prisma/schema.prisma (hunk 4 above)
Comment (BigTako) [reply prisma/schema.prisma]: Thanks for comment! Yep, number seems better type for this field. Fixed.

**Review by cherkasovnew — APPROVED** [review]

**Review by watchlar2000 — APPROVED** [review]

**Pull request merged**